// @rttist/core/dist/consts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TypeIds = exports.ModuleIds = exports.FncNames = exports.RTTIST_NAMESPACE = exports.CALLSITE_TYPE_ARGS_PROPERTY = exports.PROTOTYPE_TYPE_PROPERTY = void 0;
exports.PROTOTYPE_TYPE_PROPERTY = "[[type]]";
exports.CALLSITE_TYPE_ARGS_PROPERTY = "[[csTArgs]]";
exports.RTTIST_NAMESPACE = "Rttist";
exports.FncNames = {
    createCallsite: "cs$",
    getClassTypeParameter: "getTP$",
    invalidTypeGenerator: "utg$",
    resolveFunctionCallsite: "resFnCs$",
    resolveMethodCallsite: "resMCs$",
    getGenericClass: "getGenericClass",
    constructGeneric: "constructGeneric",
};
exports.ModuleIds = {
    Native: "::native",
    Dynamic: "::dynamic",
    Invalid: "::invalid",
    RttistType: "@rttist/dist/Type",
    RttistModule: "@rttist/dist/Module",
};
exports.TypeIds = {
    Invalid: exports.ModuleIds.Invalid + "::Invalid",
    NonPrimitiveObject: "#object",
    Function: "#Function",
    Any: "#any",
    Unknown: "#unknown",
    Void: "#void",
    Never: "#never",
    Null: "#null",
    Undefined: "#undefined",
    String: "#String",
    Number: "#Number",
    BigInt: "#BigInt",
    Boolean: "#Boolean",
    True: "#true",
    False: "#false",
    Date: "#Date",
    Error: "#Error",
    Symbol: "#Symbol",
    UniqueSymbol: "#UniqueSymbol",
    RegExp: "#RegExp",
    Int8Array: "#Int8Array",
    Uint8Array: "#Uint8Array",
    Uint8ClampedArray: "#Uint8ClampedArray",
    Int16Array: "#Int16Array",
    Uint16Array: "#Uint16Array",
    Int32Array: "#Int32Array",
    Uint32Array: "#Uint32Array",
    Float32Array: "#Float32Array",
    Float64Array: "#Float64Array",
    BigInt64Array: "#BigInt64Array",
    BigUint64Array: "#BigUint64Array",
    ArrayDefinition: "#Array",
    TupleDefinition: "#Tuple",
    ReadonlyArrayDefinition: "#ReadonlyArray",
    MapDefinition: "#Map",
    WeakMapDefinition: "#WeakMap",
    SetDefinition: "#Set",
    WeakSetDefinition: "#WeakSet",
    PromiseDefinition: "#Promise",
    GeneratorDefinition: "#Generator",
    AsyncGeneratorDefinition: "#AsyncGenerator",
    IteratorDefinition: "#Iterator",
    IterableDefinition: "#Iterable",
    IterableIteratorDefinition: "#IterableIterator",
    AsyncIteratorDefinition: "#AsyncIterator",
    AsyncIterableDefinition: "#AsyncIterable",
    AsyncIterableIteratorDefinition: "#AsyncIterableIterator",
    ArrayBuffer: "#ArrayBuffer",
    SharedArrayBuffer: "#SharedArrayBuffer",
    Atomics: "#Atomics",
    DataView: "#DataView",
};
